cli/cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,453 @@
+		c.UI.Output("Namespace: %s", ns, terminal.WithInfoStyle())
+
+		foundNamespace = ns
+	}
+
+	// Handle preset, value files, and set values logic.
+	vals, err := c.mergeValuesFlagsWithPrecedence(settings)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	valuesYaml, err := yaml.Marshal(vals)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Print out the upgrade summary.
+	if !c.flagAutoApprove {
+		c.UI.Output("Consul Upgrade Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", common.DefaultReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", foundNamespace, terminal.WithInfoStyle())
+
+		if len(vals) == 0 {
+			c.UI.Output("Overrides: "+string(valuesYaml), terminal.WithInfoStyle())
+		} else {
+			c.UI.Output("Overrides:"+"\n"+string(valuesYaml), terminal.WithInfoStyle())
+		}
+	}
+
+	// Without informing the user, default global.name to consul if it hasn't been set already. We don't allow setting
+	// the release name, and since that is hardcoded to "consul", setting global.name to "consul" makes it so resources
+	// aren't double prefixed with "consul-consul-...".
+	vals = install.MergeMaps(config.Convert(config.GlobalNameConsul), vals)
+
+	if !c.flagAutoApprove && !c.flagDryRun {
+		confirmation, err := c.UI.Input(&terminal.Input{
+			Prompt: "Proceed with upgrade? (y/N)",
+			Style:  terminal.InfoStyle,
+			Secret: false,
+		})
+
+		if err != nil {
+			c.UI.Output(err.Error(), terminal.WithErrorStyle())
+			return 1
+		}
+		if common.Abort(confirmation) {
+			c.UI.Output("Upgrade aborted. To learn how to customize your upgrade, run:\nconsul-k8s upgrade --help", terminal.WithInfoStyle())
+			return 1
+		}
+	}
+
+	if !c.flagDryRun {
+		c.UI.Output("Running Upgrade", terminal.WithHeaderStyle())
+	} else {
+		c.UI.Output("Performing Dry Upgrade", terminal.WithHeaderStyle())
+	}
+
+	// Setup action configuration for Helm Go SDK function calls.
+	actionConfig := new(action.Configuration)
+	actionConfig, err = common.InitActionConfig(actionConfig, foundNamespace, settings, uiLogger)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Setup the upgrade action.
+	upgrade := action.NewUpgrade(actionConfig)
+	upgrade.Namespace = foundNamespace
+	upgrade.DryRun = c.flagDryRun
+	upgrade.Wait = c.flagWait
+	upgrade.Timeout = c.timeoutDuration
+
+	// Read the embedded chart files into []*loader.BufferedFile.
+	chartFiles, err := common.ReadChartFiles(consulChart.ConsulHelmChart, common.TopLevelChartDirName)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Create a *chart.Chart object from the files to run the installation from.
+	chart, err := loader.LoadFiles(chartFiles)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	c.UI.Output("Downloaded charts", terminal.WithSuccessStyle())
```

Comment:
Can we change this and the similar line in install.go to "Loaded charts" instead? Since we're no longer downloading anything since the charts are embedded :)